### PR TITLE
Mautic documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added documentation for Mautic

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

**Comment by ameypv-rudder** [comment]
This PR is ready for review -`identify` and `group` snippets to be shared by @anantjain45823

**Comment by ameypv-rudder** [comment]
@anantjain45823 @shrouti1507 all the latest changes as per the spec are now included in the doc. Please review this PR and share your feedback.

**Comment by anantjain45823** [comment]
Group call can be used to link as well as delink the contacts with segment, campaign, or a company.

File: docs/destinations/streaming-destinations/mautic.mdx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+RudderStack checks if <code class="inline-code">externalId.type</code> is set to <code class="inline-code">mauticContactId</code> before using the <code class="inline-code">id</code> field for contact lookup.
+</div>
+
+<div class="warningBlock">
+The contact ID passed in your event's <code class="inline-code">externalId</code> takes precedence over the user property specified in the <strong>Mautic property name to used as lookup field</strong> dashboard setting.
+</div>
+
+## Group
+
+You can use the <Link to="/event-spec/standard-events/group/">`group`</Link> call to link a Mautic contact with a segment, campaign, or a company.
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/mautic.mdx:124]: You can use the group call to link or delink a Mautic contact with a segment, campaign, or a company.

File: docs/destinations/streaming-destinations/mautic.mdx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+A sample `group` call is shown below:
+
+```javascript
+rudderanalytics.group("group01", {
+  "name": "Alex Keener",
+  "type": "Segments",
+  "operation": "Add"
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/mautic.mdx:132]: "add" will be in lower case

File: docs/destinations/streaming-destinations/mautic.mdx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+});
+```
+
+RudderStack uses the `type` trait present in the `group` call to link a given user to a segment, campaign, or company. You can add or remove a user to a group using the `operation` trait. If this field is absent, RudderStack sets to `Add` by default.
```
Comment (anantjain45823) [inline docs/destinations/streaming-destinations/mautic.mdx:136]: RudderStack uses the `type` trait present in the `group` call to link or delink a given user to a segment, campaign, or company. You can add or remove a user to a group using the `operation` trait. If this field is absent, RudderStack sets to `add` by default.
Write Preview
Add a suggestion, <Cmd+g>
Add heading text
Add bold text, <Cmd+b>
Add italic text, <Cmd+i>
Add a quote, <Cmd+Shift+.>
Add code, <Cmd+e>
Add a link, <Cmd+k>
Add a bulleted list, <Cmd+Shift+8>
Add a numbered list, <Cmd+Shift+7>
Add a task list, <Cmd+Shift+l>
Directly mention a user or team
Reference an issue, pull request, or discussion

**Review by anantjain45823 — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Pull request merged**